jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java

```diff
@@ -940,8 +944,55 @@ private void performStorySteps(RunContext context, List<PerformableSteps> storyS
         }
 
         private void performScenarios(RunContext context) throws InterruptedException {
-            for (PerformableScenario scenario : scenarios) {
-                scenario.perform(context);
+            for (PerformableScenario performableScenario : scenarios) {
+                Meta meta = performableScenario.getScenario().getMeta();
+                if (meta.hasProperty(DEPENDS_ON_ID)) {
+                    String dependsOnId = meta.getProperty(DEPENDS_ON_ID);
+                    Boolean performDependentScenario = successfulScenariosId.get(dependsOnId);
+                    if (performDependentScenario == null || !performDependentScenario) {
+                        Throwable cause = performDependentScenario == null ? new DependsOnNotFoundException(dependsOnId)
+                                : new DependsOnFailedException(dependsOnId);
+                        UUIDExceptionWrapper wrappedCause = new UUIDExceptionWrapper(cause);
+                        Scenario scenario = performableScenario.getScenario();
+                        beforeScenario(context, scenario);
+                        if (performableScenario.hasExamples()) {
+                            context.reporter().beforeExamples(scenario.getSteps(), scenario.getExamplesTable());
+                            for (ExamplePerformableScenario exampleScenario : performableScenario.exampleScenarios) {
+                                PerformableSteps steps = exampleScenario.steps;
+                                List<Step> stepList = steps.steps;
```

Comment:
AFAIK, there also should be invocations of `StoryReporter#example`, please, review all reporting flow thoroughly